tests/subsys/net/openthread/rpc/client/commissioning/src/main.c

```diff
@@ -0,0 +1,191 @@
+	mock_nrf_rpc_tr_expect_reset();
+}
+
+/* Test serialization of otThreadDiscover() */
+ZTEST(ot_rpc_commissioning_cli, test_otThreadDiscover)
+{
+	otError error;
+	uint32_t scan_channels = 0x7fff800;
+	uint16_t pan_id = 0x4321;
+	bool joiner = false;
+	bool enable_eui64_filtering = true;
+	otHandleActiveScanResult cb = (otHandleActiveScanResult)0xdeadbeef;
+	void *cb_ctx = (void *)0xcafeface;
+
+	mock_nrf_rpc_tr_expect_add(RPC_CMD(OT_RPC_CMD_THREAD_DISCOVER, CBOR_UINT32(0x7fff800),
+					   CBOR_UINT16(0x4321), CBOR_FALSE, CBOR_TRUE, slot_cnt,
+					   CBOR_UINT32(0xcafeface)),
+				   RPC_RSP(OT_ERROR_NONE));
+	error = otThreadDiscover(NULL, scan_channels, pan_id, joiner, enable_eui64_filtering, cb,
+				 cb_ctx);
+	mock_nrf_rpc_tr_expect_done();
+	zassert_equal(error, OT_ERROR_NONE);
+}
+
+/* Test serialization of otDatasetIsCommissioned() */
+ZTEST(ot_rpc_commissioning_cli, test_otDatasetIsCommissioned)
+{
+	bool result;
+
+	mock_nrf_rpc_tr_expect_add(RPC_CMD(OT_RPC_CMD_DATASET_IS_COMMISSIONED), RPC_RSP(CBOR_TRUE));
+	result = otDatasetIsCommissioned(NULL);
+	mock_nrf_rpc_tr_expect_done();
+	zassert_true(result);
+
+	mock_nrf_rpc_tr_expect_add(RPC_CMD(OT_RPC_CMD_DATASET_IS_COMMISSIONED),
+				   RPC_RSP(CBOR_FALSE));
+	result = otDatasetIsCommissioned(NULL);
+	mock_nrf_rpc_tr_expect_done();
+	zassert_false(result);
+}
+
+/* Test serialization of otDatasetSetActiveTlvs() */
+ZTEST(ot_rpc_commissioning_cli, test_otDatasetSetActiveTlvs)
+{
+	otError error;
+
+	const otOperationalDatasetTlvs dataset = {
+		{LISTIFY(OT_OPERATIONAL_DATASET_MAX_LENGTH, LSFY_IDENTITY, (,))},
+		OT_OPERATIONAL_DATASET_MAX_LENGTH};
+
+	mock_nrf_rpc_tr_expect_add(RPC_CMD(OT_RPC_CMD_DATA_SET_ACTIVE_TLVS, TLVS),
+				   RPC_RSP(OT_ERROR_NONE));
+	error = otDatasetSetActiveTlvs(NULL, &dataset);
+	mock_nrf_rpc_tr_expect_done();
+	zassert_equal(error, OT_ERROR_NONE);
+}
+
+/* Test incoming parameters validation of otDatasetSetActiveTlvs() */
+ZTEST(ot_rpc_commissioning_cli, test_otDatasetSetActiveTlvs_negative)
+{
+	otError error;
+	const otOperationalDatasetTlvs dataset = {{0}, OT_OPERATIONAL_DATASET_MAX_LENGTH + 1};
+
+	error = otDatasetSetActiveTlvs(NULL, &dataset);
+	zassert_equal(error, OT_ERROR_INVALID_ARGS);
+	error = otDatasetSetActiveTlvs(NULL, NULL);
+	zassert_equal(error, OT_ERROR_INVALID_ARGS);
+}
+
+/* Test serialization of otDatasetGetActiveTlvs() */
+ZTEST(ot_rpc_commissioning_cli, test_otDatasetGetActiveTlvs)
+{
+	otError error;
+	otOperationalDatasetTlvs dataset;
+	const otOperationalDatasetTlvs expected_dataset = {
+		{LISTIFY(OT_OPERATIONAL_DATASET_MAX_LENGTH, LSFY_IDENTITY, (,))},
+		OT_OPERATIONAL_DATASET_MAX_LENGTH};
+
+	mock_nrf_rpc_tr_expect_add(RPC_CMD(OT_RPC_CMD_DATA_GET_ACTIVE_TLVS), RPC_RSP(TLVS));
+	error = otDatasetGetActiveTlvs(NULL, &dataset);
+	mock_nrf_rpc_tr_expect_done();
+	zassert_equal(error, OT_ERROR_NONE);
+	zassert_mem_equal(&dataset, &expected_dataset, sizeof(otOperationalDatasetTlvs));
```

Comment:
Validate length as well?

Reply:
Length is the part of `otOperationalDatasetTlvs` structure. It is checked in this comparison too.